### PR TITLE
Crate model, service, repository, connection with prop

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,6 +37,46 @@
...
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+		</dependency>
+
+
+		<dependency>
```
Comment (Denis-Balako) [inline pom.xml]: Make your dependency list more fancy - delete redundant lines 😃

File: src/main/java/application/onlinebookstore/OnlineBookStoreApplication.java [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OnlineBookStoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/OnlineBookStoreApplication.java]: Don't forget about dependency injection - provide this service through constructor and add @ Autowired above constructor.

File: src/main/java/application/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+@Data
+@Entity
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/model/Book.java]: These fields should be not null and isbn should be unique.

File: src/main/java/application/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    private String description;
+    private String coverImage;
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/model/Book.java:30]: Check the column name in DB. It responds to naming conventions?

File: src/main/java/application/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                    "SELECT b FROM Book b", Book.class).getResultList();
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
                    "FROM Book", Book.class).getResultList();
```

File: src/main/java/application/onlinebookstore/service/BookService.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package application.onlinebookstore.service;
+
+import application.onlinebookstore.model.Book;
+import java.util.List;
+
+public interface BookService {
+
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/service/BookService.java]: Redundant line

**Review by Denis-Balako — CHANGES_REQUESTED** [review]
Good job! Let's improve it!

File: src/main/java/application/onlinebookstore/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false, name = "title")
+    private String title;
+    @Column(nullable = false, name = "author")
+    private String author;
+    @Column(unique = true, name = "isbn")
+    private String isbn;
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/model/Book.java]:
```suggestion
    @Column(nullable = false, unique = true, name = "isbn")
    private String isbn;
```

File: src/main/java/application/onlinebookstore/model/Book.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    @Column(nullable = false, name = "price")
+    private BigDecimal price;
+    @Column(name = "description")
+    private String description;
+    @Column(name = "coverImage")
+    private String coverImage;
```
Comment (Denis-Balako) [inline src/main/java/application/onlinebookstore/model/Book.java]: https://medium.com/@centizennationwide/mysql-naming-conventions-e3a6f6219efe

```suggestion
    @Column(name = "cover_image")
    private String coverImage;
```

**Review by Denis-Balako — CHANGES_REQUESTED** [review]
Fix minor issues

**Review by Denis-Balako — APPROVED** [review]
Great job!

File: pom.xml [hunk 9]
```diff
@@ -32,6 +36,43 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
+
```
Comment (Vasul-Pankiv) [inline pom.xml:60]: empty line

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**